internal/factsengine/gatherers/saplocalhostresolver_test.go

```diff
@@ -0,0 +1,154 @@
+								Value: map[string]entities.FactValue{
+									"hostname": &entities.FactValueString{Value: "sapqasas"},
+									"addresses": &entities.FactValueList{
+										Value: []entities.FactValue{
+											&entities.FactValueString{Value: "10.1.1.5"},
+										},
+									},
+									"instance_name": &entities.FactValueString{Value: "ASCS00"},
+									"reachability":  &entities.FactValueBool{Value: true},
+								},
+							},
+						},
+					},
+					"NWP": &entities.FactValueList{
+						Value: []entities.FactValue{
+							&entities.FactValueMap{
+								Value: map[string]entities.FactValue{
+									"hostname": &entities.FactValueString{Value: "sapnwper"},
+									"addresses": &entities.FactValueList{
+										Value: []entities.FactValue{
+											&entities.FactValueString{Value: "10.1.1.6"},
+										},
+									},
+									"instance_name": &entities.FactValueString{Value: "ER10"},
+									"reachability":  &entities.FactValueBool{Value: false},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	suite.NoError(err)
+	suite.Equal(expectedResults, factResults)
+}
+
+func (suite *SapLocalhostResolverTestSuite) TestSapLocalhostResolverNoProfiles() {
+	appFS := afero.NewMemMapFs()
+
+	err := appFS.MkdirAll("/usr/sap/QAS", 0644)
+	suite.NoError(err)
+
+	g := gatherers.NewSapLocalhostResolver(appFS, suite.mockResolver, suite.mockPinger)
+
+	factRequests := []entities.FactRequest{{
+		Name:     "sap_localhost_resolver",
+		Gatherer: "sap_localhost_resolver",
+		CheckID:  "check1",
+	}}
+
+	factResults, err := g.Gather(factRequests)
+	suite.Nil(factResults)
+	suite.EqualError(err, "fact gathering error: saplocalhost_resolver-file-system-error - error reading the sap profiles file system: open /sapmnt/QAS/profile: file does not exist")
+}
+
+func (suite *SapLocalhostResolverTestSuite) TestSapLocalhostResolverLookupHostError() {
+	appFS := afero.NewMemMapFs()
+
+	err := appFS.MkdirAll("/usr/sap/QAS", 0644)
+	suite.NoError(err)
+
+	ascsProfileFile, _ := os.Open(helpers.GetFixturePath("gatherers/sap_profile.ascs"))
+	ascsProfileConcent, _ := io.ReadAll(ascsProfileFile)
+	err = afero.WriteFile(appFS, "/sapmnt/QAS/profile/QAS_ASCS00_sapqasas", ascsProfileConcent, 0644)
+	suite.NoError(err)
+
+	suite.mockResolver.On("LookupHost", "sapqasas").Return([]string{}, errors.New("lookup sapqasas on 169.254.169.254:53: dial udp 169.254.169.254:53: connect: no route to host"))
+
+	g := gatherers.NewSapLocalhostResolver(appFS, suite.mockResolver, suite.mockPinger)
+
+	factRequests := []entities.FactRequest{{
+		Name:     "sap_localhost_resolver",
+		Gatherer: "sap_localhost_resolver",
+		CheckID:  "check1",
+	}}
+
+	factResults, err := g.Gather(factRequests)
+
+	suite.Nil(factResults)
+	suite.EqualError(err, "fact gathering error: saplocalhost_resolver-resolution-error - error resolving hostname: lookup sapqasas on 169.254.169.254:53: dial udp 169.254.169.254:53: connect: no route to host")
```

Comment:
I find this outcome strange. Shouldn't we return an empty list of addresses here?
Otherwise, if we have 4 hostnames, and the resolution of one of them fails, the returned fact gathering would be failure, and maybe other hostnames resolve properly
PD: Commented in the prod code as well